presto-main/src/test/java/com/facebook/presto/sql/planner/TestEqualityInference.java

```diff
@@ -318,6 +335,32 @@ public void testEqualityGeneration()
         assertEquals(scopedCanonical, nameReference("a1"));
     }
 
+    @Test
+    public void testExpressionsThatMayReturnNullOnNonNullInput()
+            throws Exception
+    {
+        List<Expression> candidates = ImmutableList.of(
```

Comment:
Use a builder so things can be easily added without messing the diff. Or add an assert helper instead of using a list.